src/Data/NonEmpty.purs

```diff
@@ -54,7 +60,7 @@ oneOf :: forall f a. (Alternative f) => NonEmpty f a -> f a
 oneOf (NonEmpty a fa) = pure a <|> fa
 
 instance showNonEmpty :: (Show a, Show (f a)) => Show (NonEmpty f a) where
-  show (NonEmpty a fa) = "(NonEmpty " ++ show a ++ " " ++ show fa ++ ")"
+  show (NonEmpty a fa) = "NonEmpty (" ++ show a ++ ") (" ++ show fa ++ ")"
```

Comment:
Perhaps: 

```purescript
class (Show a) <= Atom a where
  atom :: a -> Boolean

showAtom :: forall a. (Atom a) => a -> String
showAtom a 
  | atom a = show a
  | otherwise = "(" <> show a ++ ")"
```

Then things like

```purescript
instance showNonEmpty :: (Atom a, Atom (f a)) => Show (NonEmpty f a) where
  show (NonEmpty a fa) = "NonEmpty " ++ showAtom a ++ " " ++ showAtom fa

instance atomNonEmpty :: (Atom a, Atom (f a)) => Atom (NonEmpty f a) where
  atom _ = false
```